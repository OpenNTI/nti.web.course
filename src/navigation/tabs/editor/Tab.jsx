import './Tab.scss';
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import {Input} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';

import {getTabError} from './utils';

const t = scoped('course.navigation.tabs.editor.Tab', {
	descriptions: {
		'activity': 'This tab displays a summary of social activity and content arranged on a convenient week-by-week timeline.',
		'lessons': 'This tab opens by default and contains all activities and content including readings, files, videos, assessments, discussions, and more, organized into units, lessons, and sections.',
		'scorm': 'The tab opens by default and contains the SCORM content.',
		'assignments': 'This tab contains all assignments. Learners can take assignments, view progress, and see scores. Facilitators can view learner progress, see reports, and add feedback and scores.',
		'community': 'This tab offers forums to promote engagement and discussion. Both learners and facilitators can create a new discussion. Facilitators can also post announcements.',
		'info': 'This tab includes information such as the title, description, credit awarded, start date, facilitators, and contact info. Facilitators can access reports, rosters, and other tools here.'
	}
});

export default class CourseNavigationTabsEditorTab extends React.Component {
	static propTypes = {
		tab: PropTypes.shape({
			id: PropTypes.string,
			label: PropTypes.string,
			default: PropTypes.string
		}).isRequired,
		onTabChange: PropTypes.func
	}


	onLabelChange = (label) => {
		const {tab, onTabChange} = this.props;

		if (onTabChange) {
			onTabChange(tab.id, label);
		}
	}


	render () {
		const {tab} = this.props;
		const error = getTabError(tab);

		return (
			<div className="nti-course-navigation-tabs-editor-tab">
				<div className="input-container">
					<Input.Text className={cx({error})} value={tab.label} placeholder={tab.default} onChange={this.onLabelChange} />
					<div className="error-container">{error || ''}</div>
				</div>
				<div className="description">
					{t(`descriptions.${tab.id}`)}
				</div>
			</div>
		);
	}
}
