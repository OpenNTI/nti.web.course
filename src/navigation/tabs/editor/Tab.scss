@import '~@nti/style-common';

.nti-course-navigation-tabs-editor-tab {
	display: flex;
	padding: 2rem 0;
	box-shadow: 0 1px 0 0 var(--border-grey)-light;

	.input-container {
		flex: 0 0 33%;
		margin-right: 2rem;

		input {
			width: 100%;
		}

		.error-container {
			padding: 0.3rem 0 0 0.625rem;
			font: normal 600 0.625rem/1.3 $body-font-family;
			color: var(--primary-red);
		}
	}

	.description {
		flex: 1 1 auto;
		min-height: 52px;
	}
}
