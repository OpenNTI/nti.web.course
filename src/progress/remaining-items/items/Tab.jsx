import { useCallback } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { Text } from '@nti/web-commons';

import Styles from './Style.css';

const Group = 'remaining-items-tab';

RemainingItemsTab.propTypes = {
	label: PropTypes.string,
	name: PropTypes.string,

	selected: PropTypes.bool,
	onSelect: PropTypes.func,
};
export default function RemainingItemsTab({ label, name, selected, onSelect }) {
	const onChange = useCallback(
		e => {
			if (!selected && e.target.checked) {
				onSelect?.(name);
			}
		},
		[selected, onSelect]
	);

	return (
		<label className={cx(Styles.tab, { [Styles.selected]: selected })}>
			<input
				type="radio"
				name={Group}
				checked={selected}
				onChange={onChange}
			/>
			<Text.Base>{label}</Text.Base>
		</label>
	);
}
