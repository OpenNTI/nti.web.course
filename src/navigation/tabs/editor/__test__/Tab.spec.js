/* eslint-env jest */
import TestRenderer from 'react-test-renderer';

import { Input } from '@nti/web-commons';

import Tab from '../Tab';

describe('Course Navigation Tabs Editor Tab', () => {
	test('snapshot', () => {
		const tab = { id: 'activity', label: 'Stream', default: 'Activity' };

		const testRender = TestRenderer.create(<Tab tab={tab} />);

		expect(testRender.toJSON()).toMatchSnapshot();
	});

	test('Renders a text input with appropriate value and placeholder', () => {
		const tab = {
			id: 'activity',
			label: 'Test Tab',
			default: 'Test Tab Default',
		};
		const testRender = TestRenderer.create(<Tab tab={tab} />);

		const inputCmp = testRender.root.findByType(Input.Text);

		expect(inputCmp.props.value).toEqual(tab.label);
		expect(inputCmp.props.placeholder).toEqual(tab.default);
	});

	test('Changing the input calls onTabChange with the correct arguments', () => {
		const tab = {
			id: 'activity',
			label: 'Test Tab',
			default: 'Test TabDefault',
		};
		const onTabChange = jest.fn();
		const testRender = TestRenderer.create(
			<Tab tab={tab} onTabChange={onTabChange} />
		);

		const inputCmp = testRender.root.findByType(Input.Text);

		inputCmp.props.onChange('Test Tab Update');

		expect(onTabChange).toHaveBeenCalledWith('activity', 'Test Tab Update');
	});
});
