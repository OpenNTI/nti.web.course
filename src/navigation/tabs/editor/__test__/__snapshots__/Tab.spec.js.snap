// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Navigation Tabs Editor Tab snapshot 1`] = `
<div
  className="nti-course-navigation-tabs-editor-tab"
>
  <div
    className="input-container"
  >
    <input
      className="nti-text-input"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Activity"
      type="text"
      value="Stream"
    />
    <div
      className="error-container"
    >
      
    </div>
  </div>
  <div
    className="description"
  >
    This tab displays a summary of social activity and content arranged on a convenient week-by-week timeline.
  </div>
</div>
`;
